cishouseholds/pipeline/load.py

```diff
@@ -177,14 +177,19 @@ def update_processed_file_log(df: DataFrame, filename_column: str, file_type: st
     """Collects a list of unique filenames that have been processed and writes them to the specified table."""
     spark_session = get_or_create_spark_session()
     newly_processed_files = df.select(filename_column).distinct().rdd.flatMap(lambda x: x).collect()
+    file_lengths = df.groupBy(filename_column).count().select("count").rdd.flatMap(lambda x: x).collect()
```

Comment:
This is a count of filenames. We need to get a count of the rows within each file here

Reply:
every row from the input file has a cell that contains the filename from which it originated when its added to the main dataframe therefore a count of the instances of that filename gives us the number of rows in the original file as df represents the complete df